librz/analysis/arch/v850/v850_dwarf_reg_num_table.h

```diff
@@ -0,0 +1,528 @@
+	static const char *v850_reg_names[] = { "r0", "r1", "r2", "r3", "r4", "r5", "r6", "r7",
+		"r8", "r9", "r10", "r11", "r12", "r13", "r14", "r15",
+		"r16", "r17", "r18", "r19", "r20", "r21", "r22", "r23",
+		"r24", "r25", "r26", "r27", "r28", "r29", "r30", "r31",
+		"eipc", "eipsw", "fepc", "fepsw", "ecr", "psw", "sr6", "sr7",
+		"sr8", "sr9", "sr10", "sr11", "sr12", "sr13", "sr14", "sr15",
+		"sr16", "sr17", "sr18", "sr19", "sr20", "sr21", "sr22", "sr23",
+		"sr24", "sr25", "sr26", "sr27", "sr28", "sr29", "sr30", "sr31",
+		"pc", "fp" };
+	if (regnum >= RZ_ARRAY_SIZE(v850_reg_names)) {
+		return "";
+	}
+	return v850_reg_names[regnum];
+}
+
+static const char *
+v850e_register_name(ut32 regnum) {
+	static const char *v850e_reg_names[] = {
+		"r0", "r1", "r2", "r3", "r4", "r5", "r6", "r7",
+		"r8", "r9", "r10", "r11", "r12", "r13", "r14", "r15",
+		"r16", "r17", "r18", "r19", "r20", "r21", "r22", "r23",
+		"r24", "r25", "r26", "r27", "r28", "r29", "r30", "r31",
+		"eipc", "eipsw", "fepc", "fepsw", "ecr", "psw", "sr6", "sr7",
+		"sr8", "sr9", "sr10", "sr11", "sr12", "sr13", "sr14", "sr15",
+		"ctpc", "ctpsw", "dbpc", "dbpsw", "ctbp", "sr21", "sr22", "sr23",
+		"sr24", "sr25", "sr26", "sr27", "sr28", "sr29", "sr30", "sr31",
+		"pc", "fp"
+	};
+	if (regnum >= RZ_ARRAY_SIZE(v850e_reg_names)) {
+		return "";
+	}
+	return v850e_reg_names[regnum];
+}
+
+static const char *
+v850e2_register_name(ut32 regnum) {
+	static const char *v850e2_reg_names[] = {
+		/* General purpose registers.  */
+		"r0", "r1", "r2", "r3", "r4", "r5", "r6", "r7",
+		"r8", "r9", "r10", "r11", "r12", "r13", "r14", "r15",
+		"r16", "r17", "r18", "r19", "r20", "r21", "r22", "r23",
+		"r24", "r25", "r26", "r27", "r28", "r29", "r30", "r31",
+
+		/* System registers - main banks.  */
+		"eipc", "eipsw", "fepc", "fepsw", "ecr", "psw", "pid", "cfg",
+		"", "", "", "sccfg", "scbp", "eiic", "feic", "dbic",
+		"ctpc", "ctpsw", "dbpc", "dbpsw", "ctbp", "dir", "", "",
+		"", "", "", "", "eiwr", "fewr", "dbwr", "bsel",
+
+		/* PC.  */
+		"pc", "",
+
+		/* System registers - MPV (PROT00) bank.  */
+		"vsecr", "vstid", "vsadr", "", "vmecr", "vmtid", "vmadr", "",
+		"vpecr", "vptid", "vpadr", "", "", "", "", "",
+		"", "", "", "", "", "", "", "",
+		"mca", "mcs", "mcc", "mcr",
+
+		/* System registers - MPU (PROT01) bank.  */
+		"mpm", "mpc", "tid", "", "", "", "ipa0l", "ipa0u",
+		"ipa1l", "ipa1u", "ipa2l", "ipa2u", "ipa3l", "ipa3u", "ipa4l", "ipa4u",
+		"dpa0l", "dpa0u", "dpa1l", "dpa1u", "dpa2l", "dpa2u", "dpa3l", "dpa3u",
+		"dpa4l", "dpa4u", "dpa5l", "dpa5u",
+
+		/* FPU system registers.  */
+		"", "", "", "", "", "", "fpsr", "fpepc",
+		"fpst", "fpcc", "fpcfg", "fpec", "", "", "", "",
+		"", "", "", "", "", "", "", "",
+		"", "", "", "fpspc"
+	};
+	if (regnum >= RZ_ARRAY_SIZE(v850e2_reg_names)) {
+		return NULL;
+	}
+	return v850e2_reg_names[regnum];
+}
+
+static const char *
+v850e3v5_register_name(ut32 regnum) {
+	static const char *v850e3v5_reg_names[] = {
```

Comment:
i'm ok with this, but i wonder if it would not make more sense to just have a switch with the actual values.